src/meta/src/hummock/manager/compaction_group_manager.rs

```diff
@@ -428,7 +429,8 @@ impl<S: MetaStore> HummockManager<S> {
         parent_group_id: CompactionGroupId,
         table_ids: &[StateTableId],
     ) -> Result<CompactionGroupId> {
-        self.move_state_table_to_compaction_group(parent_group_id, table_ids, None, false)
+        // 0 represents no vnode partition
+        self.move_state_table_to_compaction_group(parent_group_id, table_ids, None, false, 0)
```

Comment:
Is it better to use Option than magic number "0" ?

Reply:
> Is it better to use Option than magic number "0" ?

It's hard to use `Option` in protobuf. :lark-cry:

Reply:
```rust
pub async fn move_state_table_to_compaction_group(
        &self,
        parent_group_id: CompactionGroupId,
        table_ids: &[StateTableId],
        target_group_id: Option<CompactionGroupId>,
        allow_split_by_table: bool,
        split_weight_by_vnode: Option<u32>,
    )
```

like this ?

Reply:
> ```rust
> pub async fn move_state_table_to_compaction_group(
>         &self,
>         parent_group_id: CompactionGroupId,
>         table_ids: &[StateTableId],
>         target_group_id: Option<CompactionGroupId>,
>         allow_split_by_table: bool,
>         split_weight_by_vnode: Option<u32>,
>     )
> ```
> 
> like this ?

Yes but we cannot store `Option` in protobuf.
From another point of view, compaction needs to cut the state table every time `1/weight` of vnodes in the table have been processed. Therefore when `weight == 0`, `1/weight == inf` so we will never cut.